README.md

```diff
@@ -219,7 +231,11 @@ the default disabled, test it, and move the default from istio-system to istio-c
 
 
 ```bash
-    iop istio-control istio-autoinject $IBASE/istio-control/istio-autoinject --set enableNamespacesByDefault=true
+    # determine whether the istio-cni is installed
+    cni_pods=$(kubectl get pods -n istio-cni -l k8s-app=istio-cni-node -o "jsonpath={.items..metadata.name}")
+    
+    iop istio-control istio-autoinject $IBASE/istio-control/istio-autoinject --set enableNamespacesByDefault=true \
```

Comment:
We should have a new option 'disable iptables' or similar - since NONE interception mode is also iptable-less ( doesn't need the extra container ).

Reply:
Are you meaning we should rename `istio_cni.enabled` to something that more represents "don't include the iptables init-container"?

Reply:
Yes, absolutely. We already have this case for NONE (whitebox). 
We can leave it for separate PR - but please remove the cni_pods thing - the config generation shouldn't depend on access to k8s. If the values have it set - we use it.

What I think is critical is upgrade and gradual adoption story - we must make sure that CNI 
will work fine if some pods have injection enabled ( or use manual kube-inject ).

Reply:
Auto or manual inject doesn't matter as long as the sidecar-injector-configmap is setup correctly (with initContainer disabled).  For the case where the user wants some pods to get their iptables setup by the CNI and some to be setup by the initContainer, then multiple configmaps need to be created and manual injection should point at the appropriate configmap.  I'm not sure whether that exact scenario is clearly explained in docs.

Reply:
So if I have an existing pod, manually injected, and I add CNI - what will happen ? Will both iptables script be run ? 

Is there a documented way (annotation, etc) to disable CNI iptables ?

Reply:
NOTE:  CNI will only operate on newly created pods.  

If you have an injected manifest that has the istio-init container in it AND you have the istio-cni installed, the worst case is that pod netns iptables rules will be added by istio-cni prior to the istio-init container running and when the istio-init container runs it will clobber and re-apply all the rules.  So, unless the istio-cni and istio-init iptables setup scripts get way out of sync the user would not notice any change in resulting istio-proxy behavior.  

The istio-cni pays attention to the same annotations as the sidecar injector so, `sidecar.istio.io/inject` set to false would disable the istio-cni from doing its thing BUT it would also disable the sidecar injector.  You could toggle the value after manual inject on a pod with istio-init to avoid the istio-cni doing anything, though.  But that seems awkward.  

The istio-cni can be disabled on a namespace level in its CNI level config so that's another option unrelated to sidecar injection.

Reply:
That's great - you are right istio-init can clober the rules, so it can override. I think this is sufficient, we can improve later if anyone asks..